.travis.yml

```diff
@@ -1,13 +1,14 @@
-sudo: required
-dist: trusty
 language: python
 python:
   - 2.7
-  - 3.4
   - 3.5
   - 3.6
   # The JSON e2e test fails because, the output and exceptions are different
   # - "pypy"
+matrix:
```

Comment:
Why do we need this? Can we just put it under the `python` version, and using `xenial` for all versions?

Reply:
While xenial is _generally available_, it is not the Travis _default_ distro so I usually recommend the more conservative route of only using xenial where it is needed.  Testing both ditros is also an particular advantage on this project because it tests both paths through `script/install-linters.sh`